File: src/main/java/me/kingtux/kingserverinfo/utils/CustomArgumentUtils.java
package me.kingtux.kingserverinfo.utils;

import me.clip.placeholderapi.PlaceholderAPI;
import me.kingtux.kingserverinfo.commands.Arguments;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

public class CustomArgumentUtils {

    private static String prefix;

    public static void doArgumentWork(Arguments arguments, Player player, String prefix2) {
        prefix = prefix2;
        //Player Message
        if (arguments.getPlayerMessage() != null) {
            for (String m : arguments.getPlayerMessage()) {

                sendPlayerMessage(player, m);
            }
        }
        //Broudcast Message
        if (arguments.getBroudcastMessage() != null) {
            for (String m : arguments.getBroudcastMessage()) {
                sendBroudcastMessage(player, m);
            }
        }
        //Run player Command
        if (arguments.getPlayerCommand() != null) {
            for (String c : arguments.getPlayerCommand()) {
                runPlayerCommand(player, c);
            }
        }
        //Run Console Command
        if (arguments.getConsoleCommand() != null) {
            for (String c : arguments.getConsoleCommand()) {
                runPlayerCommand(player, c);
            }
        }


    }

    private static void sendPlayerMessage(Player p, String message) {
        message = message.replace("{prefix}", prefix);
        message = PlaceholderAPI.setPlaceholders((OfflinePlayer) p, message);
        p.sendMessage(ChatColor.translateAlternateColorCodes('&', message));
    }

    private static void sendBroudcastMessage(Player p, String message) {
        message = message.replace("{prefix}", prefix);
        message = PlaceholderAPI.setPlaceholders((OfflinePlayer) p, message);
        Bukkit.getServer().broadcastMessage(ChatColor.translateAlternateColorCodes('&', message));

    }

    private static void runConsoleCommand(Player p, String command) {
        if (p != null) {
            command = command.replace("%player_displayname%", p.getName());
            Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(), command);
        }
        Bukkit.getServer().dispatchCommand(Bukkit.getServer().getConsoleSender(), command);

    }

    private static void runPlayerCommand(Player p, String command) {
        p.performCommand(command);
    }


}

File: src/me/kingtux/kingserverinfo/utils/CustomArgumentUtils.kt
package me.kingtux.kingserverinfo.utils

import me.clip.placeholderapi.PlaceholderAPI
import me.kingtux.kingserverinfo.commands.Arguments
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.entity.Player

object CustomArgumentUtils {
    private var prefix: String? = null

    fun doArgumentWork(arguments: Arguments, player: Player, prefix2: String) {
        prefix = prefix2
        //Player Message
        if (arguments.playerMessage != null) {
            for (m in arguments.playerMessage) {

                sendPlayerMessage(player, m)
            }
        }
        //Broudcast Message
        if (arguments.broudcastMessage != null) {
            for (m in arguments.broudcastMessage) {
                sendBroudcastMessage(player, m)
            }
        }
        //Run player Command
        if (arguments.playerCommand != null) {
            for (c in arguments.playerCommand) {
                runPlayerCommand(player, c)
            }
        }
        //Run Console Command
        if (arguments.consoleCommand != null) {
            for (c in arguments.consoleCommand) {
                runPlayerCommand(player, c)
            }
        }


    }

    private fun sendPlayerMessage(p: Player, Message: String) {
        var Message = Message
        Message = Message.replace("{prefix}", prefix!!)
        Message = PlaceholderAPI.setPlaceholders(p, Message)
        p.sendMessage(ChatColor.translateAlternateColorCodes('&', Message))
    }

    private fun sendBroudcastMessage(p: Player, Message: String) {
        var Message = Message
        Message = Message.replace("{prefix}", prefix!!)
        Message = PlaceholderAPI.setPlaceholders(p, Message)
        Bukkit.getServer().broadcastMessage(ChatColor.translateAlternateColorCodes('&', Message))

    }

    private fun runConsoleCommand(p: Player?, Message: String) {
        var Message = Message
        if (p != null) {
            Message = Message.replace("%player_displayname%", p.name)
            Bukkit.getServer().dispatchCommand(Bukkit.getServer().consoleSender, Message)
        }
        Bukkit.getServer().dispatchCommand(Bukkit.getServer().consoleSender, Message)

    }

    private fun runPlayerCommand(p: Player, Message: String) {
        Bukkit.getServer().dispatchCommand(p, Message)
    }


}
